lib/rubocop/cop/lint/nested_percent_literal.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for nested percent literals.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # The percent literal for nested_attributes is parsed as four tokens,
+      #   # yielding the array [:name, :content, :"%i[incorrectly", :"nested]"].
+      #   attributes = {
+      #     valid_attributes: %i[name content],
+      #     nested_attributes: %i[name content %i[incorrectly nested]]
+      #   }
+      class NestedPercentLiteral < Cop
+        include PercentLiteral
+
+        MSG = 'Do not nest percent literals.'.freeze
+
+        # The array of regular expressions representing percent literals that,
+        # if found within a percent literal expression, will cause a
+        # NestedPercentLiteral violation to be emitted.
+        REGEXES = PercentLiteral::PERCENT_LITERAL_TYPES.map do |percent_literal|
+          /#{percent_literal}\W/
+        end.freeze
+
+        def on_array(node)
+          process(node, *PercentLiteral::PERCENT_LITERAL_TYPES)
+        end
+
+        def on_percent_literal(node)
+          add_offense(node) if contains_percent_literals?(node)
+        end
+
+        private
+
+        def contains_percent_literals?(node)
+          node.children.any? do |child|
+            literal = child.children.first.to_s
+            REGEXES.any? { |regex| literal.match(regex) }
```

Comment:
This cop  crashes with `%W[\xff]`. 

```bash
$ rubocop --cache false --only NestedPercentLiteral
An error occurred while Lint/NestedPercentLiteral cop was inspecting /tmp/tmp.zzU24B1UWS/test.rb:1:4.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while Lint/NestedPercentLiteral cop was inspecting /tmp/tmp.zzU24B1UWS/test.rb:1:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
Inspecting 1 file
.

1 file inspected, no offenses detected
```


Could you use `scrub` like https://github.com/bbatsov/rubocop/blob/6cc91fc8f0305cff3a3182934cf4b786c7f619dc/lib/rubocop/cop/layout/indent_heredoc.rb#L163

Reply:
Done (next commit).